x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiSelect, EuiButtonGroup } from '@elastic/eui';
+import { OperationDefinition } from './index';
+import { FieldBasedIndexPatternColumn } from './column_types';
+import { IndexPatternField, IndexPattern } from '../../types';
+import { updateColumnParam } from '../layer_helpers';
+import { DataType } from '../../../types';
+
+function ofName(name: string) {
+  return i18n.translate('xpack.lens.indexPattern.lastValueOf', {
+    defaultMessage: 'Last value of {name}',
+    values: { name },
+  });
+}
+
+const supportedTypes = new Set(['string', 'boolean', 'number', 'ip']);
+
+function getDateFields(indexPattern: IndexPattern): IndexPatternField[] {
+  if (indexPattern.timeFieldName) {
+    return [
+      ...indexPattern.fields.filter((field) => field.name === indexPattern.timeFieldName),
```

Comment:
not: I suspect here we're talking about a single field: if that's true, maybe `find` would better fit as it stop at the first hit.

```js
return [
      // just to be sure in case the field has been remove? Otherwise remove the array wrapping
      ...[indexPattern.fields.find( (field) => field.name === indexPattern.timeFieldName )],
	  ...indexPattern.fields.filter(
        (field) => field.type === 'date' && indexPattern.timeFieldName !== field.name
      ),
];
```